django/applications/catmaid/static/js/widgets/3dviewer.js

```diff
@@ -23,6 +23,12 @@
    * compute with web workers (betweenness centrality shading). */
   var WebGLApplication = function() {
     this.widgetID = this.registerInstance();
+
+    if (CATMAID.MAX_WEBGL_CONTEXTS - CATMAID.countWebGlContexts() < 1) {
+      CATMAID.TooManyWebGlContextsDialog().show();
```

Comment:
Is there a good pattern for aborting the instantiation of an object as I'm trying to do here? Because it doesn't init the offending widget properly, it throws more errors because the calling function tries to do things with the half-formed object.